File: learnAPI/6.r
library(jsonlite)
library(dplyr)
library(httr)
library(tibble)
library(tidyr)

url <- "https://ipinfo.io/161.185.160.93/geo"

response <- GET(url)

if (http_status(response)$category == "Success") {
    json_content <- content(response, "text")
    data <- fromJSON(json_content)

    df <- as.data.frame(data)
    print(head(df))
}

File: learnAPI/6.py
import requests
import pandas as pd

url = "https://ipinfo.io/161.185.160.93/geo"

response = requests.get(url)

if response.status_code == 200:
    data = response.json()
    
    df = pd.DataFrame({
        'Data IP':data
        })
    print(df)
else:
    print("Failed to retrieve data:", response.status_code)